File: backends/Contracts/Common.Infra/DistributedCache/Redis/Handlers/Get.cs
using D2.Contracts.Common;
using D2.Contracts.Common.App;
using S = D2.Contracts.Common.App.ICommonCacheService;

namespace Common.Infra.DistributedCache.Redis.Handlers;

/// <inheritdoc cref="S.IGetHandler{TValue}"/>
public class Get<TValue> : BaseHandler<
        S.IGetHandler<TValue>, S.GetInput, S.GetOutput<TValue>>,
    S.IGetHandler<TValue>
{
    /// <summary>
    /// Initializes a new instance of the <see cref="Get{TValue}"/> class.
    /// </summary>
    ///
    /// <inheritdoc/>
    public Get(IHandlerContext context) : base(context) { }

    /// <inheritdoc/>
    protected override ValueTask<D2Result<S.GetOutput<TValue>?>> ExecuteAsync(
        S.GetInput input,
        CancellationToken ct = default)
    {
        throw new NotImplementedException();
    }
}

File: backends/node/shared/interfaces/src/caching/in-memory/handlers/r/get.ts
import type { IHandler } from "@d2/handler";

/** Input for getting a single value from the in-memory cache. */
export interface GetInput {
  key: string;
}

/** Output for getting a single value from the in-memory cache. */
export interface GetOutput<TValue> {
  value: TValue;
}

/** Handler for getting a single value from the in-memory cache. */
export type IGetHandler<TValue> = IHandler<GetInput, GetOutput<TValue>>;
